pkg/aws/ami.py

```diff
@@ -17,86 +17,98 @@ async def start_ami_creation_worker(bucket_name, image_name, project, disk_conta
+                  "Action": "sts:AssumeRole",
+                  "Condition": {
+                     "StringEquals":{
+                        "sts:Externalid": 'vmimport'
+                     }
                   }
                }
-            }
-         ]
-      }'''
-   
-      response = client.create_role(RoleName='vmimport',AssumeRolePolicyDocument=file_trust_policy,Description='For vm migration',MaxSessionDuration=7200,Tags=[{'Key': 'app','Value': 'xmigrate'},])
-      file_role_policy = '''{
-         "Version":"2012-10-17",
-         "Statement":[
-            {
-               "Effect":"Allow",
-               "Action":[
-                  "s3:ListBucket",
-                  "s3:GetBucketLocation"
-               ],
-               "Resource":[
-                  "arn:aws:s3:::'''+bucket_name+'''"
-               ]
-            },
-            {
-               "Effect":"Allow",
-               "Action":[
-                  "s3:GetObject"
-               ],
-               "Resource":[
-                  "arn:aws:s3:::'''+bucket_name+'''/*"
+            ]
+         }
+         
+         response = client.create_role(RoleName=role_id, AssumeRolePolicyDocument=json.dumps(file_trust_policy), Description='For vm migration', MaxSessionDuration=7200, Tags=[{'Key': 'app','Value': 'xmigrate'}])
+         print(f'Created role {role_id}')
+
+         file_role_policy = {
+               "Version":"2012-10-17",
+               "Statement":[
+                  {
+                     "Effect": "Allow",
+                     "Action": [
+                        "kms:CreateGrant",
+                        "kms:Decrypt",
+                        "kms:DescribeKey",
+                        "kms:Encrypt",
+                        "kms:GenerateDataKey*",
+                        "kms:ReEncrypt*"
+                     ],
+                     "Resource": "*"
+                  },
+                  {
+                     "Effect": "Allow",
+                     "Action": [
+                        "license-manager:GetLicenseConfiguration",
+                        "license-manager:UpdateLicenseSpecificationsForResource",
+                        "license-manager:ListLicenseSpecificationsForResource"
+                     ],
+                     "Resource": "*"
+                  }
                ]
-            },
-            {
-               "Effect":"Allow",
-               "Action":[
-                  "ec2:ModifySnapshotAttribute",
-                  "ec2:CopySnapshot",
-                  "ec2:RegisterImage",
-                  "ec2:Describe*"
-               ],
-               "Resource":"*"
             }
-         ]
-      }'''
-      response = client.put_role_policy(
-         RoleName='vmimport',
-         PolicyName='vmimport',
-         PolicyDocument=file_role_policy
-      )
+         response = client.put_role_policy(
+            RoleName= role_id,
+            PolicyName= 'vmimport',
```

Comment:
here too..

Reply:
I can change this one but I thought since the inline policy is always the same a common name is better.